internal/services/azurestackhci/stack_hci_cluster_data_source.go

```diff
@@ -132,3 +158,21 @@ func (r StackHCIClusterDataSource) Read() sdk.ResourceFunc {
 		},
 	}
 }
+
+func flattenSystemAssignedToModel(input *identity.SystemAndUserAssignedMap) []identity.ModelSystemAssigned {
```

Comment:
Is there a reason we can't use the existing flatten functions in the `identity` package?

Reply:
The service API only supports systemAssigned Identity, which means there should be no `identity_ids` field in the tf schema, but the swagger and SDK uses `identity.SystemAndUserAssignedMap`, so need mannually skip the `identity_ids` in expand/flatten function.

Reply:
Then we should raise an issue on the swagger since the definition doesn't align with the API behaviour and link it here. Can you please do that?

Reply:
change this in swagger would be considered as breaking change, and the user assigned identity might be supported in the future, can we leave it for now, and remove this if user assigned identity is supported later.

Reply:
@teowa we should always raise an issue on the swagger spec when what is defined in the spec does not correspond to the behaviour of the API, which is the case here. Whether the service team actually make the breaking change to fix it or not is another question, this is about tracking for when the service is brought in line. Please open an issue on the swagger spec and link it here, like we do for all other discrepancies.

Reply:
@stephybun thanks for clarifying , I thought I need to modify the swagger. The issue has been raised: https://github.com/Azure/azure-rest-api-specs/issues/28260